lib/std/wasm.zig

```diff
@@ -231,6 +234,445 @@ test "Wasm - valtypes" {
     testing.expectEqual(@as(u8, 0x7C), _f64);
 }
 
+/// Wasm ops, but without input/output/signedness information
+/// Used for `buildOpcode`
+pub const Op = enum {
+    @"unreachable",
+    nop,
+    block,
+    loop,
+    @"if",
+    @"else",
+    end,
+    br,
+    br_if,
+    br_table,
+    @"return",
+    call,
+    call_indirect,
+    drop,
+    select,
+    local_get,
+    local_set,
+    local_tee,
+    global_get,
+    global_set,
+    load,
+    store,
+    memory_size,
+    memory_grow,
+    @"const",
+    eqz,
+    eq,
+    ne,
+    lt,
+    gt,
+    le,
+    ge,
+    clz,
+    ctz,
+    popcnt,
+    add,
+    sub,
+    mul,
+    div,
+    rem,
+    @"and",
+    @"or",
+    xor,
+    shl,
+    shr,
+    rotl,
+    rotr,
+    abs,
+    neg,
+    ceil,
+    floor,
+    trunc,
+    nearest,
+    sqrt,
+    min,
+    max,
+    copysign,
+    wrap,
+    convert,
+    demote,
+    promote,
+    reinterpret,
+    extend,
+};
+
+/// Contains the settings needed to create an `Opcode` using `buildOpcode`.
+///
+/// The fields correspond to the opcode name. Here is an example
+///          i32_trunc_f32_s
+///          ^   ^     ^   ^
+///          |   |     |   |
+///   valtype1   |     |   |
+///     = .i32   |     |   |
+///              |     |   |
+///             op     |   |
+///       = .trunc     |   |
+///                    |   |
+///             valtype2   |
+///               = .f32   |
+///                        |
+///                width   |
+///               = null   |
+///                        |
+///                   signed
+///                   = true
+///
+/// There can be missing fields, here are some more examples:
+///   i64_load8_u
+///     --> .{ .valtype1 = .i64, .op = .load, .width = 8, signed = false }
+///   i32_mul
+///     --> .{ .valtype1 = .i32, .op = .trunc }
+///   nop
+///     --> .{ .op = .nop }
+pub const OpcodeBuildArguments = struct {
+    /// First valtype in the opcode (usually represents the type of the output)
+    valtype1: ?Valtype = null,
+    /// The operation (e.g. call, unreachable, div, min, sqrt, etc.)
+    op: Op,
+    /// Width of the operation (e.g. 8 for i32_load8_s, 16 for i64_extend16_i32_s)
+    width: ?u8 = null,
+    /// Second valtype in the opcode name (usually represents the type of the input)
+    valtype2: ?Valtype = null,
+    /// Signedness of the op
+    signedness: ?Signedness = null,
+};
+
+/// Helper function that builds an Opcode given the arguments needed
+pub fn buildOpcode(args: OpcodeBuildArguments) Opcode {
```

Comment:
These "opinionated" functions should go into src/codegen/wasm.zig rather than into the standard library. What goes in the std lib is "bits" - that is - just enums and names for magic numbers. `buildOpcode` and related types contains logic that is not inherent to wasm, but is related to a compiler backend.

Reply:
Done. moved `buildOpcode` to src/codegen/wasm.zig as suggested.